scrape/manager.go

```diff
@@ -146,6 +146,8 @@ func (m *Manager) DroppedTargets() []*Target {
 }
 
 func (m *Manager) reload(t map[string][]*targetgroup.Group) {
+	m.mtx.Lock()
```

Comment:
Is it really necessary to acquire a lock for the whole duration of the reload? The `Sync()` calls use their own locks already.

Reply:
the lock is for the `m.scrapePools` map
write at 
https://github.com/krasi-georgiev/prometheus/blob/e564bc0a88321b275afbc0bdcae565b342c7dc0d/scrape/manager.go#L162
while read at
https://github.com/krasi-georgiev/prometheus/blob/e564bc0a88321b275afbc0bdcae565b342c7dc0d/scrape/manager.go#L124